mozci/util/hgmo.py

```diff
@@ -36,20 +38,66 @@ def __init__(self, rev, branch="autoland"):
     @staticmethod
     def create(rev, branch="autoland"):
         key = (branch, rev[:12])
-        if key in HGMO.CACHE:
-            return HGMO.CACHE[key]
-        instance = HGMO(rev, branch)
-        HGMO.CACHE[key] = instance
+        if key in HgRev.CACHE:
+            return HgRev.CACHE[key]
+        instance = HgRev(rev, branch)
+        HgRev.CACHE[key] = instance
         return instance
 
-    def _get_resource(self, url):
+    @staticmethod
+    def _get_and_cache_pushes(branch, url):
```

Comment:
Nit: can you add type information to the new functions you added? It should be simple in these cases

Reply:
Oops, yes I need to get in the habit more.